bfvmm/src/vmcs/test/test_vmcs_intel_x64.cpp

```diff
@@ -20,3 +21,183 @@
 // Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
 
 #include <test.h>
+#include <vmcs/vmcs_intel_x64.h>
+#include <memory_manager/memory_manager.h>
+
+static std::map<uint32_t, uint64_t> g_msrs;
+static std::map<uint64_t, uint64_t> g_vmcs_fields;
+
+static uint64_t
+read_msr(uint32_t msr)
+{
+    return g_msrs[msr];
+}
+
+static void
+write_msr(uint32_t msr, uint64_t val)
+{
+    g_msrs[msr] = val;
+}
+
+static bool
+vmread(uint64_t field, uint64_t *val)
+{
+    *val = g_vmcs_fields[field];
+    return true;
+}
+
+static bool
+vmwrite(uint64_t field, uint64_t val)
+{
+    g_vmcs_fields[field] = val;
+    return true;
+}
+
+static uint16_t
+get_16bit_state()
+{
+    return 0;
+}
+
+static uint32_t
+get_32bit_state()
+{
+    return 0;
+}
+
+static uint64_t
+get_64bit_state()
+{
+    return 0;
+}
+
+static void
+dump()
+{
+}
+
+static uintptr_t
+virt_to_phys_ptr(void *ptr)
+{
+    (void) ptr;
+
+    return 0x0000000ABCDEF0000;
+}
+
+static void
+setup_vmcs_x64_state_intrinsics(MockRepository &mocks, vmcs_intel_x64_state *state_in)
+{
+    // Setup 16 bit state functions
+    mocks.OnCall(state_in, vmcs_intel_x64_state::es).Do(get_16bit_state);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::cs).Do(get_16bit_state);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::ss).Do(get_16bit_state);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::ds).Do(get_16bit_state);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::ds).Do(get_16bit_state);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::fs).Do(get_16bit_state);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::gs).Do(get_16bit_state);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::ldtr).Do(get_16bit_state);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::tr).Do(get_16bit_state);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::gdt_limit).Do(get_16bit_state);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::idt_limit).Do(get_16bit_state);
+
+    // Setup 32 bit state functions
+    mocks.OnCall(state_in, vmcs_intel_x64_state::es_limit).Do(get_32bit_state);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::cs_limit).Do(get_32bit_state);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::ss_limit).Do(get_32bit_state);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::ds_limit).Do(get_32bit_state);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::ds_limit).Do(get_32bit_state);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::fs_limit).Do(get_32bit_state);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::gs_limit).Do(get_32bit_state);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::ldtr_limit).Do(get_32bit_state);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::tr_limit).Do(get_32bit_state);
+
+    mocks.OnCall(state_in, vmcs_intel_x64_state::es_access_rights).Do(get_32bit_state);
```

Comment:
each one of these needs it's own function which will be used later on in tests to be added in the future.